extensions/amp-form/0.1/test/test-amp-form.js

```diff
@@ -539,36 +542,37 @@ describes.repeated('', {
     });
 
     it('should allow rendering responses through inlined templates', () => {
-      return getAmpForm(getForm(env.win.document, true)).then(ampForm => {
-        const form = ampForm.form_;
-        // Add a div[submit-error] with a template child.
-        const errorContainer = document.createElement('div');
-        errorContainer.setAttribute('submit-error', '');
-        form.appendChild(errorContainer);
-        const errorTemplate = document.createElement('template');
-        errorTemplate.setAttribute('type', 'amp-mustache');
-        errorTemplate.content.appendChild(
-            document.createTextNode('Error: {{message}}'));
-        errorContainer.appendChild(errorTemplate);
-        let renderedTemplate = document.createElement('div');
-        renderedTemplate.innerText = 'Error: hello there';
-        sandbox.stub(ampForm.xhr_, 'fetch').returns(Promise.reject({
-          response: {
-            status: 400,
-            json() {
-              return Promise.resolve({message: 'hello there'});
-            },
+      const ampForm = getUnresolvedAmpForm(getForm(env.win.document, true));
+      const form = ampForm.form_;
+      // Add a div[submit-error] with a template child.
+      const errorContainer = document.createElement('div');
+      errorContainer.setAttribute('submit-error', '');
+      form.appendChild(errorContainer);
+      const errorTemplate = document.createElement('template');
+      errorTemplate.setAttribute('type', 'amp-mustache');
+      errorTemplate.content.appendChild(
+          document.createTextNode('Error: {{message}}'));
+      errorContainer.appendChild(errorTemplate);
+      let renderedTemplate = document.createElement('div');
+      renderedTemplate.innerText = 'Error: hello there';
+      sandbox.stub(ampForm.xhr_, 'fetch').returns(Promise.reject({
+        response: {
+          status: 400,
+          json() {
+            return Promise.resolve({message: 'hello there'});
           },
-        }));
-        sandbox.stub(ampForm.templates_, 'findAndRenderTemplate')
-            .returns(Promise.resolve(renderedTemplate));
-        const event = {
-          stopImmediatePropagation: sandbox.spy(),
-          target: form,
-          preventDefault: sandbox.spy(),
-        };
-        ampForm.handleSubmitEvent_(event);
-        const findTemplateStub = ampForm.templates_.findAndRenderTemplate;
+        },
+      }));
+      sandbox.stub(ampForm.templates_, 'findAndRenderTemplate')
+          .returns(Promise.resolve(renderedTemplate));
+      const event = {
+        stopImmediatePropagation: sandbox.spy(),
+        target: form,
+        preventDefault: sandbox.spy(),
+      };
+      ampForm.handleSubmitEvent_(event);
+      const findTemplateStub = ampForm.templates_.findAndRenderTemplate;
+      allowConsoleError(() => {
```

Comment:
This has the same problem I mentioned in chat. The returned promise won't be chained (unless this has been since fixed).

Reply:
Noted. Not yet fixed.

Reply:
I'm not sure what you're referring to

Reply:
Place a breakpoint on line 577. I believe it won't get triggered.

This is because sinon chains returned promises, but `allowConsoleError()` currently swallows it. What we need is for `allowConsoleError(f)` to detect promises returned in by `f` and chain them. Then, we'll do:

```js
return allowConsoleError(() => { 
  ...
});
```

Reply:
Fixed in #15228. With this, you can return `allowConsoleError`. Improperly written tests will fail presubmit checks.